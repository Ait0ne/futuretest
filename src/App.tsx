import React from 'react';
import DataSizePicker from './components/DataSizePicker/dataSize-picker.component'
import {useUsersList} from './components/UsersListProvider/users-list-provide.component';
import Loader from './components/Loader/loader.component';
import Main from './components/Main/main.component';
import Alert from './components/Alert/alert.component';

import {AppContainer} from './App.styles';

const  App = () => {
  const {loading, error, usersList, addUser, clearErrors}  = useUsersList()

  return (
      <AppContainer>
        {
          loading? <Loader/>
          :null
        }
        <DataSizePicker />
        {
          error? <Alert onClose={clearErrors} message='При загрузке данных произошла ошибка'/>
          :null
        }
        {
          usersList.length>0 ? <Main list={usersList} addRow={addUser}/> : null
        }
      </AppContainer>
  );
}

export default App;
