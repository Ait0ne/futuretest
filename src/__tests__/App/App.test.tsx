import React from 'react';
import {mount, shallow} from 'enzyme';
import App from '../../App';
import {UsersListContext, UsersListProvider, UserListContextProps} from '../../components/UsersListProvider/users-list-provide.component';

export const ListContextTestProps:UserListContextProps = {
    loading: true,
    error: false,
    addUser: jest.fn(),
    usersList: [],
    fetchUsersList: jest.fn(),
    clearErrors: jest.fn(),
}

describe('app in loading state', () => {
    const app = mount(
    <UsersListContext.Provider
    value={ListContextTestProps}
    >
        <App/>
    </UsersListContext.Provider>
    )
    it('shows loader', () => {
        expect(app.find('Loader').length).toEqual(1)
    } )
})

describe('app in normal state', () => {
    const app = mount(<App/>, {
        wrappingComponent: UsersListProvider
    })
    
    it('renders correctly', () => {
        expect(app).toMatchSnapshot()
    })

    it('renders DataSizePicker', () => {
        expect(app.find('DataSizePicker').length).toEqual(1)
    })

    it("doesn't show Loader", () => {
        expect(app.find('Loader').length).toEqual(0)
    })
})

describe('app in error state', () => {
    const app = mount(
        <UsersListContext.Provider
        value={{...ListContextTestProps, error:true}}
        >
            <App/>
        </UsersListContext.Provider>
        )

    it('renders alert component', () => {
        expect(app.find('Alert').length).toEqual(1)
    })
})