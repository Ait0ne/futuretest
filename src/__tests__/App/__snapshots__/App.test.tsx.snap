// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app in normal state renders correctly 1`] = `
<App>
  <styled.div>
    <div
      className="sc-fFSRdu dJGnaS"
    >
      <DataSizePicker>
        <styled.div>
          <div
            className="sc-dlnjPT hqyxOp"
          >
            <span>
              Выберите размер набора данных:
            </span>
            <styled.div>
              <div
                className="sc-hKFyIo eNOVOQ"
              >
                <styled.button
                  disabled={false}
                  name="smallDataSize"
                  onClick={[Function]}
                >
                  <button
                    className="sc-bdnylx kJaVHl"
                    disabled={false}
                    name="smallDataSize"
                    onClick={[Function]}
                  >
                    Маленький
                  </button>
                </styled.button>
                <styled.button
                  disabled={false}
                  name="largeDataSize"
                  onClick={[Function]}
                >
                  <button
                    className="sc-bdnylx kJaVHl"
                    disabled={false}
                    name="largeDataSize"
                    onClick={[Function]}
                  >
                    Большой
                  </button>
                </styled.button>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </DataSizePicker>
    </div>
  </styled.div>
</App>
`;
